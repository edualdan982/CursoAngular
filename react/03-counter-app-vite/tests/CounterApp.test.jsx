const { render, screen, fireEvent } = require("@testing-library/react");
const { default: CounterApp } = require("../src/CounterApp");

describe("Prueba en <CounterApp/>", () => {
  test("debe hacer match con el snapshot", () => {
    // expect(render.debug()).toMatchSnapshot();

    const { container } = render(<CounterApp />);
    expect(container).toMatchSnapshot();
  });

  const initialValue = 100;
  test("debe de mostrar el valor inicial de 100", () => {
    render(<CounterApp value={initialValue} />);

    expect(screen.getByTestId("test-counter").innerHTML).toBe(
      initialValue.toString()
    );

    // ----tambien es valido buscar el valor 100 porque este deberia existir en algun lado del redenderizado

    // expect(screen.getByText(100)).toBeTruthy();

    // ----otra forma es buscar a nivel del h3 en mi caso con:

    // expect(screen.getByRole('heading', {level: 3}).innerHTML).toContain('100');
  });

  test("debe de incrementar con el boton +1", () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText("+1"));

    expect(screen.getByText(initialValue + 1)).toBeTruthy();
  });

  test("debe de decrementar con el boton -1", () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText("-1"));

    expect(screen.getByText(initialValue - 1)).toBeTruthy();
  });

  test("debe funcionar el boton de reset", () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));

    // fireEvent.click(screen.getByText("reset"));

    // fireEvent.click(screen.getByRole("button", { name : 'reset' }));
    fireEvent.click(screen.getByTestId("test-btn-reset"));
    expect(screen.getByTestId("test-counter").innerHTML).toBe(
      initialValue.toString()
     );
  });
});
