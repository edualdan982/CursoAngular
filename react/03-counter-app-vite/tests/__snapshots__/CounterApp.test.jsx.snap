// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <CounterApp/> debe hacer match con el snapshot 1`] = `
<div>
  <h1 />
  <h2>
    Contador:
  </h2>
  <h3
    data-testid="test-counter"
  >
    0
  </h3>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    data-testid="test-btn-reset"
    id="btn-reset"
  >
    reset
  </button>
</div>
`;
